### PR TITLE
LMS Performance Fixes

### PR DESCRIPTION
**Pull request by Jeremyyang920**
This PR makes several changes to help increase performance by decreasing the amount of GC that is required by reusing memory and allocations as much as possible.

### CODE REVIEW

File: server/resp.go [hunk 1]
```diff
@@ -150,18 +167,69 @@ func (c *Conn) writeResultset(r *Resultset) error {
...
+	bufferedData = c.bufferEOF(bufferedData)
+	if _, err := c.Write(bufferedData); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (c *Conn) bufferFieldList(in []byte, fs []*Field, data []byte) []byte {
+	/*
```
Comment (otan) [inline server/resp.go]: let's remove the commented out sections

File: client/resp.go [hunk 2]
```diff
@@ -406,20 +406,22 @@ func (c *Conn) readResultRows(result *Result, isBinary bool) (err error) {
 
 		result.RowDatas = append(result.RowDatas, data)
 	}
-
+	/*
```
Comment (otan) [inline client/resp.go]: let's undo this change for now. we should make a builtin to fill in `results.Values` and call it everytime.

**Comment by otan** [comment]
is there a way to make all the tests pass?

**Comment by Jeremyyang920** [comment]
> is there a way to make all the tests pass?

Let me see. I think theres something weird with the github action. Since It has no output.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -6,8 +6,9 @@ jobs:
     strategy:
       matrix:
         go: [ 1.19, 1.18, 1.17, 1.16 ]
-    name: Tests Go ${{ matrix.go }}
-    runs-on: ubuntu-18.04
+        os: [ ubuntu-22.04, ubuntu-20.04 ]
+    name: Tests Go ${{ matrix.go }} # This name is used in main branch protection rules
+    runs-on: ${{ matrix.os }}
```
Comment (Jeremyyang920) [inline .github/workflows/ci.yml:11]: I coped this from the main repo since the old ubuntu-18.04 is deprecated for golang tests now and thats why the actions never ran.
